flocker/volume/filesystems/zfs.py

```diff
@@ -67,26 +71,100 @@ def connectionLost(self, reason):
         del self._result
 
 
-def zfs_command(reactor, arguments):
+class _AsyncLZC(object):
```

Comment:
This sounds a little bit too advanced for my skills.
I can try to do that or it can be deferred to a more skilled developer.
In either case I will work on this item and see what I can do. At the very least I'll add documentation.